src/types.ts

```diff
@@ -60,7 +62,7 @@ export type InheritedProps<T> = WithoutUseProps<
  * Props of a component created with `use()`
  * @template T The type of the `use` prop
  */
-export type UseProps<T> = React.Props<any> &
+export type UseProps<T> = React.ClassAttributes<T> &
```

Comment:
Hmm, I'm not sure if this will work properly. `T` is passed to the `ref` prop, right? Maybe it should be `any`?
```suggestion
export type UseProps<T> = React.ClassAttributes<any> &
```